stor/dx.py

```diff
@@ -1,32 +1,41 @@
 from cached_property import cached_property
-import six
+from contextlib2 import contextmanager
+import logging
+import sys
+import tempfile
+import warnings
 
 import dxpy
-import dxpy.exceptions as dx_exceptions
+from dxpy.exceptions import DXSearchError
 from dxpy.exceptions import DXError
+import six
 
 from stor import exceptions as stor_exceptions
 from stor import Path
+from stor import settings
 from stor import utils
 from stor.obs import OBSPath
+from stor.obs import OBSUploadObject
+from stor.posix import PosixPath
 
 
 DNAnexusError = stor_exceptions.RemoteError
 NotFoundError = stor_exceptions.NotFoundError
+ConditionNotMetError = stor_exceptions.ConditionNotMetError
+ConflictError = stor_exceptions.ConflictError
+UnavailableError = stor_exceptions.UnavailableError
+UnauthorizedError = stor_exceptions.UnauthorizedError
+TargetExistsError = stor_exceptions.TargetExistsError
 
+logger = logging.getLogger(__name__)
+progress_logger = logging.getLogger('%s.progress' % __name__)
 
-def _parse_dx_error(exc, **kwargs):
-    """
-    Parses DXError exceptions to throw a more informative exception.
-    """
-    msg = exc.message
-    exc_type = type(exc)
-
-    if exc_type is dx_exceptions.DXSearchError:
-        if msg and 'found more' in msg.lower():
-            return DuplicateError(msg, exc)
-        elif msg and 'found none' in msg.lower():
-            return NotFoundError(msg, exc)
+auth_token = settings.get()['dx']['auth_token']
```

Comment:
No, only if the environment variable is set obviously. If it's not set, dxpy will look for an `environment.json` file in `~/{USER}/.dnanexus_config/` folder (which is the default config folder). If it doesn't find anything there, it will raise an error like `dxpy.exceptions.PermissionDenied: VIEW permission required in project-xxxx to perform this action, code 401.` which gets hoisted accordingly.